openmdao/solvers/linear/direct.py

```diff
@@ -57,11 +57,11 @@ def format_singular_error(err, system, mtx):
 
     n = 0
     varname = "Unknown"
-    for name in system._var_allprocs_abs_names['output']:
-        relname = system._var_abs2prom['output'][name]
-        n += len(system._outputs[relname])
-        if loc <= n:
-            varname = relname
+    varsizes = system._var_sizes['nonlinear']['output']
+    for j, name in enumerate(system._var_allprocs_abs_names['output']):
+        n += varsizes[0][j]
```

Comment:
might want to use owning_rank here instead of 0.  Var might not exist on proc 0.

Reply:
Modified this to index via owning_rank.